### DISCUSSION TITLE
Mask UID value problem

### DISCUSSION THREAD

**Discussion by simakvladimir** [uNmAnNeR/imaskjs#1003] [Q&A]
Hello everyone!

IMask tool is awesome, but I have some problem with correctly tuning it for UID value. 

Example - https://codepen.io/simakvladimir/pen/Jjzwxyj

All works as expected me except such moment:
- start print `1` when input is empty. Value in input is changed to `10000000-0000-0000-0000-000000000000`
- move caret at `end-1` pos - `10000000-0000-0000-0000-00000000000|0` and print `2` - `10000000-0000-0000-0000-000000000002|`
- move caret before 5th character - `1000|0000-0000-0000-0000-000000000002`
- press "delete" key on keyboard, value changed to -  `1000|2000-0000-0000-0000-000000000000`, but desired/expected value - `1000|0000-0000-0000-0000-000000000020`

How I can achieve desired behavior? I don't want shift value to first unfilled placeholder symbol position. I want value shifts on one char to left when delete pressed.

Thanks in advance!

**Comment by uNmAnNeR** [comment]
Hi @simakvladimir, you are welcome.
I don't think it's possible out of the box. The only thing you can do is to subclass from pattern mask and override `nearestInputPos` method. In fact, you need to understand how one of the most complicated part in the lib works.

**Reply by uNmAnNeR** [reply]
and you can use backquote (`) to prevent symbols shift back

**Reply by simakvladimir** [reply]
1. Ok, I try deep dive to pattern mask impl
2. I've played with backquote before, but I couldn't get the behavior I wanted from them

Thanks for reply and have a good day!
